third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -564,13 +564,26 @@ struct LoadOpConversion
 
     unsigned numRepOuter = numReps[opIdx];
     unsigned numRepInner = numReps[!opIdx];
+
+    Value pitch = trunc(i32_ty, rowStride);
+    if (!memoryRowMajor) {
+      // Column major memory. We need to swap the width and height because HW
+      // only support row major memory layout.
+      pitch = trunc(i32_ty, colStride);
+      std::swap(baseWidth, baseHeight);
+    }
```

Comment:
```suggestion
    Value pitch;
    if (memoryRowMajor) {
      pitch = trunc(i32_ty, rowStride);
    } else {
      // Column major memory. We need to swap the width and height because HW
      // only support row major memory layout.
      pitch = trunc(i32_ty, colStride);
      std::swap(baseWidth, baseHeight);
    }
```

Avoid creating two operations.